### PR TITLE
Update libcxx and libcxxabi to LLVM 10 release branch

### PR DESCRIPTION
**Pull request by dschuff**
All of the local diffs applied cleanly to 10.x.
test_emscripten_scan_registers and test_emscripten_scan_stack were updated to directly #include <assert.h> which was apparently being included indirectly via libcxx, and now isn't.
test_metadce_cxx has been rebaseline.

### CODE REVIEW

**Comment by dschuff** [comment]
Unfortunately this didn't solve #11022 but I decided to go ahead and finish it up anyway.

File: system/include/libcxx/__config [hunk 1]
```diff
@@ -343,9 +339,26 @@
...
+#    if __ANDROID_API__ >= 21
+#      define _LIBCPP_HAS_QUICK_EXIT
+#    endif
+#    if __ANDROID_API__ >= 28
+#      define _LIBCPP_HAS_ALIGNED_ALLOC
+#    endif
+#    if __ANDROID_API__ >= 29
+#      define _LIBCPP_HAS_TIMESPEC_GET
+#    endif
+#  elif defined(__Fuchsia__) || defined(__wasi__) || defined(__NetBSD__)
```
Comment (sbc100) [inline system/include/libcxx/__config:360]: I wonder if we should add emscripten here too.. not part of this change though.

File: system/lib/libcxxabi/include/__cxxabi_config.h [hunk 2]
```diff
@@ -71,7 +71,8 @@
 #define _LIBCXXABI_NO_CFI
 #endif
 
-#if defined(__arm__)
+// wasm32 follows the arm32 ABI convention of using 32-bit guard.
+#if defined(__arm__) || defined(__wasm32__)
```
Comment (sbc100) [inline system/lib/libcxxabi/include/__cxxabi_config.h:75]: Nice.. its a shame we probably still need to define this in `tools/system_libs.py` because fastcomp.

I guess you could update the TODO in `tools/system_libs.py` and wrap it in a check for fastomp instead?

File: system/lib/libcxx/readme.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
-These files are from libc++, release 9.0.0.
+These files are from libc++, release branch 10.x.
 
-tag: llvmorg-9.0.0
-git: 0399d5a9682b3cef71c653373e38890c63c4c365
+tag: none
```
Comment (sbc100) [inline system/lib/libcxx/readme.txt:3]: Is there no tag that you sync'd to?  Wouldn't it be better to sync to a given tag?

Comment (sbc100) [reply system/lib/libcxx/readme.txt:3]: There is separate readme for libcxxabi

Comment (dschuff) [reply system/lib/libcxx/readme.txt:3]: The current revision is past the 10.0 release tag, but since all of the commits on the release branch are bugfixes and such, I didn't really see any reason to leave any out?
Fixed the libcxxabi readme in the next commit.

Comment (sbc100) [reply system/lib/libcxx/readme.txt:3]: hmm.. still they are as-yet-unreleased changes.    Why not just stick with the tag.   No need to go back and change now, we can aim for tag next time.

Comment (dschuff) [reply system/lib/libcxx/readme.txt:3]: Turns out none of the changes on the branch touched libcxx or libcxxabi, we can just pretend it came from the tagged revision.

File: system/lib/libcxxabi/include/__cxxabi_config.h (hunk 2 above)
Comment (dschuff) [reply system/lib/libcxxabi/include/__cxxabi_config.h:75]: how can we check for fastcomp in system_libs.py?
Or I guess we could just update the comment so that when we grep for fastcomp later we find it?

Comment (sbc100) [reply system/lib/libcxxabi/include/__cxxabi_config.h:75]: `Settings.WASM_BACKEND`

Comment (sbc100) [reply system/lib/libcxxabi/include/__cxxabi_config.h:75]: !
